<launch>
    <arg name="rviz" default="false" />
    <arg name="obstacle_angle_thrsh" default="0.785398" />
    
    <node
        pkg="lidar_obstacle_detection"
        name="lidar_obstacle_detection_node"
        type="lidar_obstacle_detection_node"
        output="screen">
    <param name="obstacle_angle_thrsh" type="double" value="$(arg obstacle_angle_thrsh)"/>
    </node>
    
    <group if="$(arg rviz)">
        <node 
            launch-prefix="nice" 
            pkg="rviz" 
            type="rviz" 
            name="rviz_obstacles" 
            args="-d $(find lidar_obstacle_detection)/rviz/obstacle_visualization.rviz" 
        />
    </group>

</launch>
